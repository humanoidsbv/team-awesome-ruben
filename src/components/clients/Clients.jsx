import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import Client from '../client/';
import ClientAdd from '../client-add/';
import IconArrowDown from '../../assets/icons/icon-arrow-down.svg';
import IconPlus from '../../assets/icons/icon-plus.svg';
import styles from './Clients.module.css';

const Clients = ({ addClient, clients, fetchClients }) => {
  const [isFormVisible, SetIsFormVisible] = useState(false);

  useEffect(function getClients() {
    fetchClients();
  }, []);

  const handleFormVisibility = () => SetIsFormVisible(!isFormVisible);

  return (
    <React.Fragment>
      {isFormVisible && (
        <ClientAdd
          addClient={addClient}
          toggleFormVisibility={handleFormVisibility}
        />
      )}
      <div className={styles.header}>
        <span className={styles.heading}>All Clients</span>
        <button
          className={`${styles.addMember}
            ${isFormVisible && styles.addMemberVisible}`}
          disabled={isFormVisible}
          onClick={handleFormVisibility}
          type="button"
        >
          <IconPlus className={styles.addNewIcon} />
          New Client
        </button>
        <button type="button" className={styles.sortMembers}>
          Sort by:
          <IconArrowDown />
        </button>
      </div>
      {clients.map(client => (
        <React.Fragment key={client.id}>
          <Client client={client} />
        </React.Fragment>
      ))}
    </React.Fragment>
  );
};

Clients.propTypes = {
  addClient: PropTypes.func.isRequired,
  clients: PropTypes.arrayOf(
    PropTypes.shape({
      client: PropTypes.string
    })
  ),
  fetchClients: PropTypes.func.isRequired
};

Clients.defaultProps = {
  clients: []
};

export default Clients;
