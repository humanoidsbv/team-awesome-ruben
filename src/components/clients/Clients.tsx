import React, { useState, useEffect } from 'react';

import Client from '../client';
import ClientAdd from '../client-add';
import IconArrowDown from '../../assets/icons/icon-arrow-down.svg';
import IconPlus from '../../assets/icons/icon-plus.svg';
import styles from './Clients.module.css';

import { ClientsPropsInterface } from '../../redux/clients/types';

const Clients = ({
  addClient,
  clients,
  fetchClients
}: ClientsPropsInterface): React.ReactElement => {
  const [isFormVisible, SetIsFormVisible] = useState(false);

  useEffect(() => {
    fetchClients();
  }, []);

  const handleFormVisibility = () => SetIsFormVisible(!isFormVisible);

  return (
    <React.Fragment>
      {isFormVisible && (
        <ClientAdd
          addClient={addClient}
          data-test="ClientAdd"
          toggleFormVisibility={handleFormVisibility}
        />
      )}
      <div className={styles.header}>
        <span className={styles.heading}>All Clients</span>
        <button
          className={`${styles.addMember} ${
            isFormVisible ? styles.addMemberVisible : ''
          }`}
          disabled={isFormVisible}
          onClick={handleFormVisibility}
          type="button"
          data-test="addMember"
        >
          <IconPlus className={styles.addNewIcon} />
          New Client
        </button>
        <button type="button" className={styles.sortMembers}>
          Sort by:
          <IconArrowDown />
        </button>
      </div>
      {clients.map(client => (
        <Client client={client} key={client.id} />
      ))}
    </React.Fragment>
  );
};

export default Clients;
