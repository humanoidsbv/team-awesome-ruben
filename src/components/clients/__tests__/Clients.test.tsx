import * as React from 'react';
import { shallow } from 'enzyme';

import Clients from '../Clients';

const mockedClientsData = {
  branch: '',
  branchAddress: '',
  companyName: '',
  id: 0,
  locality: '',
  postalCode: ''
};

it('matches component output with expected output', () => {
  const clientAdd = shallow(
    <Clients
      addClient={() => {}}
      clients={[mockedClientsData]}
      fetchClients={() => {}}
    />
  );
  expect(clientAdd).toMatchSnapshot();
});
