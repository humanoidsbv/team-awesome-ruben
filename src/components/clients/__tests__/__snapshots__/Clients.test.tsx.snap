// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component output with expected output 1`] = `
<Fragment>
  <div
    className="header"
  >
    <span
      className="heading"
    >
      All Clients
    </span>
    <button
      className="addMember
            false"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <test-file-stub
        className="addNewIcon"
      />
      New Client
    </button>
    <button
      className="sortMembers"
      type="button"
    >
      Sort by:
      <test-file-stub />
    </button>
  </div>
  <Client
    client={
      Object {
        "branch": "",
        "branchAddress": "",
        "companyName": "",
        "id": 0,
        "locality": "",
        "postalCode": "",
      }
    }
    key="0"
  />
</Fragment>
`;
